pkg/authenticator/authenticator.go

```diff
@@ -254,6 +264,35 @@ func (auth *Authenticator) ParseAuthenticationResponse(response []byte) error {
 	return nil
 }
 
+// We define these functions so we can mock WaitForInjectedCertificateFile
+type OsStatFunc func(name string) (os.FileInfo, error)
+type OsIsNotExistFunc func(err error) bool
+
+// WaitForInjectedCertificateFile waits for retryCountLimit seconds to see if the cert file
+// exists in the given path. If it's not there by the end of the retry count limit, it returns
+// an error.
+func (auth *Authenticator) WaitForInjectedCertificateFile(certificatePath string, retryCountLimit int, osStatFunc OsStatFunc, osIsNotExistFunc OsIsNotExistFunc) error {
+	constantBackOff := backoff.NewConstantBackOff(time.Second)
+	count := 0
+	return backoff.Retry(func() error {
+		info, err := osStatFunc(certificatePath)
+		if !osIsNotExistFunc(err) && !info.IsDir() {
```

Comment:
I know this is old code but a check for "is it a file" seems more appropriate than "is it not a dir": https://golang.org/pkg/os/#FileMode.IsRegular